### PR TITLE
Update docs with new session key information

### PR DESCRIPTION
**Pull request by JoaoSa-da**

### CODE REVIEW

**Review by marciw-da — APPROVED** [review]
Approving but please note suggestions and one critical fix (build error)

File: docs/2.9.0/docs/canton/usermanual/performance.rst [hunk 1]
```diff
@@ -171,7 +171,7 @@ Typically `X` only contains the receiver of the transfer, but not the contract a
 
 Then, we observe the following storage usage:
 
-- Two encrypted envelopes with payload `Y` each, one symmetric key per
+- Two encrypted envelopes with payload `Y` each, two symmetric keys per
```
Comment (soren-da) [inline docs/2.9.0/docs/canton/usermanual/performance.rst:174]: this is slightly outdated, right? 1) view seed per view, 2) symmetric key per informee group

File: docs/2.9.0/docs/canton/usermanual/security.rst [hunk 2]
```diff
@@ -149,10 +149,27 @@ A transaction is composed of multiple views due to sub-transaction privacy.
...
+
+To avoid having to asymmetrically decrypt the view encryption key for all
+transactions we make use of a short-lived session key that acts as an intermediary encryption layer positioned
+between the view encryption and the encryption with the participants' keys.
+The process involves encrypting and decrypting the session key (that is used to encrypt the view encryption key)
+using the participants' public and private keys, with the resulting data cached temporarily in memory.
+This caching mechanism enables similar transactions involving the same participants to make use of the cached
+information, thereby acquiring the session key without necessitating asymmetric key operations for each transaction.
+
+:warning: Reusing session keys between requests sacrifices forward secrecy, but enhances overall performance. For optimal security, we suggest disabling caching in exchange of performance improvements.
```
Comment (soren-da) [inline docs/2.9.0/docs/canton/usermanual/security.rst:167]: As discussed please rephrase

File: docs/2.9.0/docs/canton/architecture/overview.rst [hunk 3]
```diff
@@ -411,16 +411,24 @@ To that end, the submitting participant encrypts the views using the following h
    each such symmetric key is used only once.
 
 #. It encrypts the serialization of each view's Merkle tree with the symmetric key derived for this view.
-   The view seed itself is encrypted with the public key of each participant hosting an informee of the view.
-   The encrypted Merkle tree and the encryptions of the view seed form the data that is sent via the sequencer to the recipients.
+   The view seed itself is encrypted with a short-lived symmetric session key that is generated for each distinct
+   recipient informee group (set of participants that will receive the view). Finally this session key is encrypted
+   with the public key of each participant hosting an informee of the view. If caching is enabled, the
+   asymmetric encryption/decryption results are briefly stored in memory, to
```
[the remark this answers is not in the document]
Comment (JoaoSa-da) [reply docs/2.9.0/docs/canton/architecture/overview.rst:417]: I rephrased this and now I mention the session key...

**Review by soren-da — APPROVED** [review]

**Comment by garyverhaegen-da** [comment]
We have [started the 3.1 documentation folder "for real"](https://github.com/digital-asset/docs.daml.com/pull/675). This pull requests makes changes to 2.9.0; please consider whether this should be forward-ported to 3.1 before merging it.

File: docs/3.1/versions.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "daml": "2.8.0-snapshot.20231118.12382.0.v86cb8054",
-  "canton": "2.9.0-snapshot.20231219.11655.0.v11e7e1e8",
+  "canton": "2.9.0-snapshot.20240402.11920.0.v2507c677",
```
Comment (garyverhaegen-da) [inline docs/3.1/versions.json:3]: We'll probably want a 3.1 snapshot here at some point.

**Review by garyverhaegen-da — APPROVED** [review]

**Pull request merged**